daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1637,24 +1642,29 @@ private[lf] object SBuiltin {
         crash(s"value not a list of parties or party: $v")
     }
 
+  private[this] val keyWithMaintainersStructFields: Struct[Unit] =
+    Struct.assertFromSeq(List(keyFieldName -> (()), maintainersFieldName -> (())))
+
+  private[this] val keyIdx = keyWithMaintainersStructFields.indexOf(keyFieldName)
+  private[this] val maintainerIdx = keyWithMaintainersStructFields.indexOf(maintainersFieldName)
+
   private[this] def extractKeyWithMaintainers(
       v: SValue,
   ): Node.KeyWithMaintainers[V[Nothing]] =
     v match {
-      case SStruct(flds, vals)
-          if flds.length == 2 && flds(0) == Ast.keyFieldName && flds(1) == Ast.maintainersFieldName =>
+      case SStruct(flds, vals) if flds.names sameElements keyWithMaintainersStructFields.names =>
```

Comment:
Do we really need this `sameElements` runtime check here? We're in full control of the code producing the structural record, aren't we?